### PR TITLE
Feature/filter movies

### LINKED ISSUE #9: As a user, I can search for movies, and see the movies that match the details
- There should be a searchbar on landing page. 
- The searchbar should have placeholder text that guides user to search
- Upon each keystroke, the movies should be filtered
- There should be a clear icon that removes search keyword and shows all movies again. 
- Search filtering should be case-insensitive

Scenario: User types substring
Acceptance criteria: All movies that either match the title or the substring, should show up.

Scenario: User removing characters
Acceptance criteria: filter should still re-render and show more movies that match the revised keyword

Scenario: User clicked cross icon
Acceptance criteria: All movies should show up and searchbar should be cleared

### PR DESCRIPTION
**Pull request by taranveersingh93**
#### What is the feature?
- Searchbar on homepage that filters movie by substrings and a cancel search
- If no movies found, error message is displayed
- The search is case-insensitive

#### How you implemented the solution?
- Used controlled components. Followed data down actions up approach. 
- Error message is displayed using conditional rendering

#### Any changes in the UI (Screenshots)?
-
Screenshot 2023-07-07 at 9 26 43 PM

#### How to test the feature (A test scenario or any new setup)?
Scenario: User types substring
Acceptance criteria: All movies that either match the title or the substring, should show up.

Scenario: User removing characters
Acceptance criteria: filter should still re-render and show more movies that match the revised keyword

Scenario: User clicked cross icon
Acceptance criteria: All movies should show up and searchbar should be cleared

#### Any other notes:
- none

### CODE REVIEW

File: src/components/AllMovies/AllMovies.js [hunk 1]
```diff
@@ -17,11 +16,19 @@ const AllMovies = ({ movies }) => {
     )
   })
 
+  const ErrorMessage = () => {
+    return (<h3 className='error-message'>Sorry, No movies to display</h3>);
```
Comment (paulina-isabel) [inline src/components/AllMovies/AllMovies.js:20]: Nice error handling.

File: src/components/AllMovies/FilterRibbon/FilterRibbon.css [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  cursor: pointer;
+  transform: translateY(-2px);
+  transition: 0.3s ease;
+}
+
+.cross-container img {
+  height: 100%;
+  filter: invert();
+  opacity: 0.7;
+}
```
Comment (paulina-isabel) [inline src/components/AllMovies/FilterRibbon/FilterRibbon.css:38]: I love the styling! 💪

**Review by paulina-isabel — APPROVED** [review]
Cool job with the clearing search functionality. Sweet!

File: src/components/App/App.js [hunk 3]
```diff
@@ -4,20 +4,32 @@ import Navbar from '../Navbar/Navbar';
...
   const [onDetailsView, setOnDetailsView] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
   const [movies, setMovies] = useState(movieData.movies);
 
+  const filterMovies = (keyword, movieList) => {
+    return [...movieList].filter(movie => movie.title.toLowerCase().includes(keyword.toLowerCase()))
+  }
+
+  const changeSearch = value => {
+    console.log(searchValue, "before")
```
Comment (niksseif) [inline src/components/App/App.js:22]: Make sure to remove console before pushing up code.

File: src/components/App/App.js [hunk 4]
```diff
@@ -4,20 +4,32 @@ import Navbar from '../Navbar/Navbar';
...
+    console.log(searchValue, "after")
+    setMovies(filterMovies(value, movieData.movies));
+  }
+  
   return (
     <div className="App">
       <Navbar />
       <main>
-        {onHomeView && <AllMovies movies={movies}/>}
+        {onHomeView && <AllMovies changeSearch={changeSearch} movies={movies} searchValue={searchValue}/>}
```
Comment (niksseif) [inline src/components/App/App.js:32]: Question: What happens when the movies are not defined? I noticed that you have handled the error in the AllMovies component, but is it also possible to handle it here?

**Review by niksseif — COMMENTED** [review]
Nice work 🙌🏼

**Pull request merged**